proxy/src/main/java/com/velocitypowered/proxy/command/VelocityCommandManager.java

```diff
@@ -142,7 +150,21 @@ public void unregister(final String alias) {
     Preconditions.checkNotNull(alias, "alias");
     // The literals of secondary aliases will preserve the children of
     // the removed literal in the graph.
-    dispatcher.getRoot().removeChildByName(alias.toLowerCase(Locale.ENGLISH));
+    String aliasLowercase = alias.toLowerCase(Locale.ENGLISH);
+    dispatcher.getRoot().removeChildByName(aliasLowercase);
+
+    CommandMeta meta = commandMetas.get(alias);
+    if (meta != null) {
+      for (String metaAlias : meta.getAliases()) {
+        commandMetas.remove(metaAlias);
```

Comment:
Nit: perhaps this should check if the meta is the same as the primary alias with `Map.remove(key, onlyIfValue)`.